# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/ipc/WebrtcTCPSocket.cpp
# Commit: c43589fd0f90
# Full Hash: c43589fd0f90d174eb185b563406fd0a2e6ea712
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Update OnConnected callbacks to specify what type of proxying (if any) is being used, and implement IsProxied based on this information. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45186
# ==============================================================================

diff -r 940d7f787bea -r c43589fd0f90 media/mtransport/ipc/WebrtcTCPSocket.cpp
--- a/media/mtransport/ipc/WebrtcTCPSocket.cpp	Tue Sep 17 18:18:15 2019 +0000
+++ b/media/mtransport/ipc/WebrtcTCPSocket.cpp	Tue Sep 17 18:18:33 2019 +0000
@@ -231,17 +231,16 @@
   nsresult rv;
 
   if (NS_SUCCEEDED(aResult) && aProxyinfo) {
-    nsCString proxyType;
-    rv = aProxyinfo->GetType(proxyType);
+    rv = aProxyinfo->GetType(mProxyType);
     if (NS_WARN_IF(NS_FAILED(rv))) {
       CloseWithReason(rv);
       return rv;
     }
 
-    if (proxyType == "http" || proxyType == "https") {
+    if (mProxyType == "http" || mProxyType == "https") {
       rv = OpenWithHttpProxy();
-    } else if (proxyType == "socks" || proxyType == "socks4" ||
-               proxyType == "direct") {
+    } else if (mProxyType == "socks" || mProxyType == "socks4" ||
+               mProxyType == "direct") {
       rv = OpenWithoutHttpProxy(aProxyinfo);
     }
   } else {
@@ -452,7 +451,7 @@
 
   MOZ_ASSERT(mProxyCallbacks, "webrtc TCP callback should be non-null");
 
-  mProxyCallbacks->OnConnected();
+  mProxyCallbacks->OnConnected(mProxyType);
 }
 
 void WebrtcTCPSocket::InvokeOnRead(nsTArray<uint8_t>&& aReadData) {